app/scripts/lib/metrics.js

```diff
@@ -121,6 +121,7 @@ define(function (require, exports, module) {
 
     this._marketingImpressions = {};
     this._activeExperiments = {};
+    this._eventMemory = [];
```

Comment:
> Another alternative is to skip storing a list of events locally and search this.events directly:

I don't think we can do that because those get flushed so it won't be logged once. 
Otherwise an Object makes sense 👍